src/components/Carousel.js

```diff
@@ -443,10 +479,24 @@ class Carousel extends Component {
           position = this.props.infiniteLoop ? 0 : lastPosition;
         }
 
-        this.selectItem({
-            // if it's not a slider, we don't need to set position here
-            selectedItem: position
-        });
+        if (needClonedSlide) {
+            this.selectItem({
+                selectedItem: position,
+                loop: true
+            });
+            requestAnimationFrame(() => {
```

Comment:
This also doesn't smell right... why would you need 2 `raf` here?

Reply:
Because I need two `setState`s in two frames. One `rAF` will schedule setState at the end of current frame, which react will batch them together.

Reply:
This seems too fragile. If something delays rendering you might have a race condition here.

React provides a callback to setState which only gets called after the state is applied. It should work for your use case. You can change `selectItem` to receive the callback function and pass it down to setState like:
```
selectItem = (state, callback) => {
    this.setState(state, callback);
    ...
}
```

And then here, you change your code to:

```
this.selectItem({
    selectedItem: position,
    loop: true
}, () => {
    this.selectItem({
        loop: false
    });
});
```

Reply:
I still don't think it's a good solution. I made such change and published a demo [here](react-responsive-carousel.surge.sh) (second story) and you can see it in action: it doesn't work. 

Due to using microtask for scheduling, `setState` callback may schedule state change in the same frame. However, the race condition is a valid concern too. So I think it is safe to use nested `rAF` inside a `setState` callback, this way we get rid of the race condition and make sure two animation renders in two frames. (working [demo](react-responsive-carousel-inifinte-scrolling.surge.sh))

(Both have sourcemap enabled so you can inspect source directly)

Reply:
@driskell I'm surly comment in codes with above reasoning if we land on this implementation.